### PR TITLE
feat(babel): typescript builder

### PR DESCRIPTION
**Pull request by xhyrom**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const { AsyncQueue } = require('../lib/index');
```
Comment (AngeloCore) [inline test/test.js]: remove `test.js`, we don't need it

File: test/test.ts [hunk 2]
```diff
@@ -1,37 +1,37 @@
-import { AsyncQueue } from "./src";
```
Comment (AngeloCore) [inline test/test.ts]: rename to `__tests__/index.ts`

File: tsconfig.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
     "compilerOptions": {
         "removeComments": false,
         "module": "commonjs",
-        "target": "ES6",
+        "target": "ES2021",
```
Comment (AngeloCore) [inline tsconfig.json]: use nodejs v14

**Review by AngeloCore — CHANGES_REQUESTED** [review]

File: test/test.js (hunk 1 above)
Comment (xhyrom) [reply test/test.js]: ok

**Review by AngeloCore — APPROVED** [review]

**Pull request merged**